R/test-server.R

```diff
@@ -3,6 +3,27 @@ isModuleServer <- function(x) {
   is.function(x) && names(formals(x))[1] == "id"
 }
 
+#' @noRd
+isAppDir <- function(path) {
+
+  if (file.exists(file.path(path, "app.R")))
+    return(TRUE)
+
+  if (file.exists(file.path(path, "server.R"))
+      && file.exists(file.path(path, "ui.R")))
+    return(TRUE)
+
+  FALSE
+}
+
+#' @noRd
+findEnclosingApp <- function(path = ".") {
+  rprojroot::find_root(
+    rprojroot::root_criterion(isAppDir, "is a Shiny app"),
+    path
+  )
+}
```

Comment:
Can this function also be set for the default of loadSupport?

```r
loadSupport <- function(appDir = findEnclosingApp("."), ....
```

Reply:
I don't know if it would be needed to be exported, but it wouldn't hurt if it is set as a default value.